op-chain-ops/deployer/state/state.go

```diff
@@ -22,16 +26,60 @@ type State struct {
 	// deployment. It only contains the proxies because the implementations
 	// can be looked up on chain.
 	SuperchainDeployment *SuperchainDeployment `json:"superchainDeployment"`
+
+	// ImplementationsDeployment contains the addresses of the common implementation
+	// contracts required for the Superchain to function.
+	ImplementationsDeployment *ImplementationsDeployment `json:"implementationsDeployment"`
+
+	// Chains contains data about L2 chain deployments.
+	Chains []ChainState `json:"opChainDeployments"`
 }
 
 func (s State) WriteToFile(path string) error {
 	return jsonutil.WriteJSON(s, ioutil.ToAtomicFile(path, 0o755))
 }
 
 type SuperchainDeployment struct {
-	ProxyAdminAddress            common.Address `json:"proxyAdminAddress"`
-	SuperchainConfigProxyAddress common.Address `json:"superchainConfigProxyAddress"`
-	SuperchainConfigImplAddress  common.Address `json:"superchainConfigImplAddress"`
-	ProtocolVersionsProxyAddress common.Address `json:"protocolVersionsProxyAddress"`
-	ProtocolVersionsImplAddress  common.Address `json:"protocolVersionsImplAddress"`
+	ProxyAdminAddress            common.Address       `json:"proxyAdminAddress"`
+	SuperchainConfigProxyAddress common.Address       `json:"superchainConfigProxyAddress"`
+	SuperchainConfigImplAddress  common.Address       `json:"superchainConfigImplAddress"`
+	ProtocolVersionsProxyAddress common.Address       `json:"protocolVersionsProxyAddress"`
+	ProtocolVersionsImplAddress  common.Address       `json:"protocolVersionsImplAddress"`
+	StateDump                    *foundry.ForgeAllocs `json:"stateDump"`
```

Comment:
You could embed the struct of the `opsm` deployer into this struct (ends up mixing up the API layer, but would reduce duplication).

Also, comment about the `stateDump`: good to keep this separated from deployers. We can avoid all the intermediate state-dumps/loads during op-e2e when we execute a full pipeline of steps on top of the same host.